src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -135,7 +136,15 @@ def remove_partition(device):
     :param device: A ``Device()`` object
     """
     udev_info = udevadm_property(device.path)
-    partition_number = udev_info.get('ID_PART_ENTRY_NUMBER')
+    # Sometimes there's a race condition that makes 'ID_PART_ENTRY_NUMBER' be not present
+    # in the output of `udevadm info --query=property`.
+    # Probably not ideal and not the best fix but this allows to get around that issue.
+    # The idea is to make it retry multiple times before actually failing.
+    for i in range(10):
+        partition_number = udev_info.get('ID_PART_ENTRY_NUMBER', '1')
```

Comment:
I think you need to call `udevadm_property()` in this loop too?

Reply:
![Alt Text](https://c.tenor.com/vhdL79GE0fsAAAAC/its-true-the-office.gif)

Reply:
![image](https://i.giphy.com/media/abittW9YasaxSimk0E/giphy.webp)